src/pages/index.js

```diff
@@ -19,64 +21,187 @@ import {
+    name: data.title,
+    link: data.url,
+  };
+
+  addCardPopup.setLoading("Saving...");
+
+  api
+    .addCards(cardData)
+    .then((res) => {
+      const card = createCard(res);
+      cardList.addItem(card);
+      addCardFormElement.reset();
+      addFormValidator.resetValidation();
+      addCardPopup.close();
+      addCardPopup.setLoading("Save");
+    })
+    .catch((err) => {
+      console.error(`Error, could not add card: ${err}`);
+    })
+    .finally(() => {
+      addCardFormElement.reset();
+      addFormValidator.resetValidation();
+      addCardPopup.setLoading("Save");
+    });
+});
+
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
-const editFormValidator = new FormValidator(settings, profileEditForm);
-const addFormValidator = new FormValidator(settings, addCardFormElement);
+function handleLikeClick(data) {
+  if (!data.isLiked) {
+    api
+      .likeACard(data._id, { method: "PUT" })
+      .then((res) => {
+        console.log(res);
+        data.isLiked = true;
+        data.toggleLike();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .dislikeACard(data._id, { method: "DELETE" })
+      .then((res) => {
+        console.log(res);
+        data.isLiked = false;
+        data.toggleLike();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+function handleDeleteClick(card) {
+  const cardId = card._id;
+  if (!cardId) {
+    console.error("card._id is undefined");
+    return;
+  }
+  confirmDeleteModal.setSubmitFunction(() => {
+    api
+      .handleDeleteCard(cardId)
+      .then((res) => {
+        console.log(res);
+        card.handleDeleteCard();
+        confirmDeleteModal.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+  confirmDeleteModal.open();
+}
 
 function createCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleCardPreview);
+  const cardElement = new Card(
+    cardData,
+    "#card-template",
+    handleCardPreview,
+    handleLikeClick,
+    handleDeleteClick
+  );
   return cardElement.getView();
 }
 
 previewImagePopup.setEventListeners();
 
-const editProfilePopup = new PopupWithForm(
-  "#edit-modal",
-  handleProfileEditSubmit
-);
+const editProfilePopup = new PopupWithForm("#edit-modal", (userData) => {
+  editProfilePopup.setLoading("Saving...");
+  handleProfileEditSubmit(userData);
+});
 editProfilePopup.setEventListeners();
 
 addCardPopup.setEventListeners();
 
-cardList.renderItems();
+// cardList.renderItems();
 
 function handleCardPreview(cardData) {
   previewImagePopup.open(cardData);
 }
 
 // Event Handlers
 
-function handleProfileEditSubmit({ name, description }) {
-  userInfo.setUserInfo({ name, description });
+function handleProfileEditSubmit(userData) {
+  const { name, description } = userData;
```

Comment:
According to the task you need to change submit buttons text (`Saving…`) while waiting for the server response. And the user should see it.